### PR TITLE
Refactor: Use `$escaper` instead of deprecated `$block` escapers

### PR DESCRIPTION
**Pull request by sanganinamrata**
## Purpose of this pull request

Refactor: Use `$escaper` instead of deprecated `$block` escapers in devdoc examples as a part of Magento 2 coding standards improvements. I've made these updates as a part of https://devdocs.magento.com/guides/v2.4/extension-dev-guide/xss-protection.html#output-processing documentation.

## Affected DevDocs pages

- [Custom widget](https://devdocs.magento.com/guides/v2.3/ext-best-practices/tutorials/custom-widget.html)
- [View models](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/view-models.html)
- [Custom validation rules](https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/validations/custom-validation.html)
- [RowBuilder widget](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget-row-builder.html)
- [QuickSearch widget](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_quickSearch.html)
- [HtmlContent component](https://devdocs.magento.com/guides/v2.3/ui_comp_guide/components/ui-htmlcontent.html)

## Links to Magento source code

N/A

### CODE REVIEW

File: src/guides/v2.3/extension-dev-guide/view-models.md [hunk 1]
```diff
@@ -133,9 +133,10 @@ The following is an example of the view model initialization in the `app/code/Ma
 
 ```php
 /** @var $viewModel /Magento/Catalog/ViewModel/Product/Listing/PreparePostData */
+/** @var $escaper-> \Magento\Framework\Escaper */
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/view-models.md:136]: Should this not be -
```suggestion
/** @var $escaper \Magento\Framework\Escaper */
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]

File: src/guides/v2.3/extension-dev-guide/view-models.md (hunk 1 above)
Comment (sanganinamrata) [reply src/guides/v2.3/extension-dev-guide/view-models.md]: @BarnyShergold Your suggested changes are done.

**Review by BarnyShergold — APPROVED** [review]
All good! Thanks for your contribution

**Comment by sanganinamrata** [comment]
@BarnyShergold Will you please proceed this PR?

**Comment by keharper** [comment]
@sanganinamrata When was this code deprecated?

**Comment by dobooth** [comment]
@sanganinamrata

**Comment by dobooth** [comment]
Closing as abandoned. Please create a PR if you still want these changes made. Thanks.

**Pull request closed without merging**